designs/odcr.md

```diff
@@ -0,0 +1,213 @@
+      - [All Launch Templates are associated with the specified Capacity Reservation](#all-launch-templates-are-associated-with-the-specified-capacity-reservation)
+      - [Pricing and consolidation](#pricing-and-consolidation)
+        - [Provisioning](#provisioning)
+        - [Consolidating Capacity Reserved Instances](#consolidating-capacity-reserved-instances)
+        - [Consolidating into Capacity Reserved Instances](#consolidating-into-capacity-reserved-instances)
+      - [Labels](#labels)
+    - [Failed to launch Nodes into Capacity Reservation](#failed-to-launch-nodes-into-capacity-reservation)
+
+## Background
+
+In AWS [ODCR](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-capacity-reservations.html) allows users to reserve compute capacity to mitigate the risk of 
+getting on-demand capacity. This is very helpful during seasonal holidays where higher traffic is expected or for reserving highly-desired instance types, like the
+`p5.48xlarge` or other large GPU instance types.
+
+### Capacity Reservations
+
+Each [Capacity Reservation](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2@v1.162.1/types#CapacityReservation) is defined with:
+
+- The Availability Zone in which to reserve the capacity
+- The number of instances for which to reserve capacity
+- The instance attributes, including the instance type, tenancy, and platform/OS
+- Instance match criteria
+  - Targeted -- only accept instances that matches all attributes + explicitly targeted the capacity reservation
+  - Open -- if capacity reservation accepts all instances that matches all attributes
+- A start and end date (if applicable) for when the reservation of capacity is available
+
+AWS also supports grouping Capacity Reservation into Capacity Reservation groups. 
+Both these entities are supported in Launch Template's CapacityReservationTarget [definitions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-capacityreservationtarget.html).
+
+## Goals
+
+- Support associating ODCRs to EC2NodeClass
+  - open
+  - targeted
+- Define Karpenter's behavior when configuring Capacity Reservations for an EC2NodeClass
+- Define Karpenter's behavior when encountering errors when attempting to launch nodes into Capacity Reservation
+- Define Karpenter's behavior when consolidating nodes to Capacity Reservation when available
+
+## Non-Goals
+
+- Supporting capacity-blocks as a capacity-type.
+- Supporting changes in scaling behavior when ODCR is associated to a NodeClass. _We won't bring up N nodes to match an N node capacity reservation, this would directly interfear with the ability to share a Capacity Reservation between multiple clusters or accounts. The first Karpenter finding the Capacity Reservation would provision all instances for a reservation leaving nothing unused_
+- Supporting Capacity Reservation Groups. _Adding this abstraction for now adds additional complexity_
```

Comment:
I wonder if it's not even about adding complexity. It's a natural extension of this design. We'd allow for CRGs which would cause us to select all ODCRs in a CRG when we do discovery